wpimath/src/main/java/edu/wpi/first/math/geometry/Quaternion.java

```diff
@@ -192,31 +212,45 @@ private Quaternion _exp() {
     var axial_magnitude = Math.sqrt(getX() * getX() + getY() * getY() + getZ() * getZ());
     // cos(||v||)
     var cosine = Math.cos(axial_magnitude);
-    // sin(||v||)
-    var sine = Math.sin(axial_magnitude);
+
+    double axial_scalar;
+
+    if (axial_magnitude < 1e-9) {
+      // Taylor series of sin(x) near x=0: 1 - x^2 / 6 + x^4 / 120 + O(n^6)
+      axial_scalar = 1 - Math.pow(axial_magnitude, 2) / 6 + Math.pow(axial_magnitude, 4) / 120;
```

Comment:
Should the 2 calls to `Math.pow` be replaced with 2 double multiplications? (1 to get `axial_magnitude_sq`, then another to get the 4th power)
(Also in C++)

Reply:
It should. Math.pow(double, double) doesn't have a special case for y=4: https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/lang/FdLibm.java#L2116